packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,259 @@ export class IamPolicyChecksWebview extends VueWebview {
+                )
+                return
+            case 'Terraform Plan':
+                const tfCommand = `tf-policy-validator validate --template-path ${document} --region ${this.region} --config ${defaultTerraformConfigPath}`
+                this.executeValidatePolicyCommand(tfCommand)
+                return
+            case 'CloudFormation':
+                const cfnCommand =
+                    cfnParameterPath === ''
+                        ? `cfn-policy-validator validate --template-path ${document} --region ${this.region}`
+                        : `cfn-policy-validator validate --template-path ${document} --region ${this.region} --template-configuration-file ${cfnParameterPath}`
+                this.executeValidatePolicyCommand(cfnCommand)
+                return
+        }
+    }
+
+    public async checkNoNewAccess(
+        documentType: string,
+        policyType: string,
+        referenceDocument: string,
+        cfnParameterPath?: string
+    ) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        customPolicyCheckDiagnosticCollection.clear()
+        if (referenceDocument !== '') {
+            fs.writeFileSync(tempFilePath, referenceDocument)
+        } else {
+            this.onCustomPolicyCheckResponse.fire(['Reference policy document is missing.', 'red'])
+            return
+        }
+
+        switch (documentType) {
+            case 'Terraform Plan':
+                const tfCommand = `tf-policy-validator check-no-new-access --template-path ${document} --region ${this.region} --config ${defaultTerraformConfigPath} --reference-policy ${tempFilePath} --reference-policy-type ${policyType}`
+                this.executeCustomPolicyChecksCommand(tfCommand)
+                return
+            case 'CloudFormation':
+                const cfnCommand =
+                    cfnParameterPath === ''
+                        ? `cfn-policy-validator check-no-new-access --template-path ${document} --region ${this.region} --reference-policy ${tempFilePath} --reference-policy-type ${policyType}`
+                        : `cfn-policy-validator check-no-new-access --template-path ${document} --region ${this.region} --reference-policy ${tempFilePath} --reference-policy-type ${policyType} --template-configuration-file ${cfnParameterPath}`
+                this.executeCustomPolicyChecksCommand(cfnCommand)
+                return
+        }
+        if (fs.existsSync(tempFilePath)) {
+            fs.unlinkSync(tempFilePath)
+        }
+    }
+
+    public async checkAccessNotGranted(documentType: string, referenceDocument: string, cfnParameterPath?: string) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        customPolicyCheckDiagnosticCollection.clear()
+        if (referenceDocument !== '') {
+            // Remove spaces, line breaks, carriage returns, and tabs
+            referenceDocument = referenceDocument.replace(/\s*|\t|\r|\n/gm, '')
```

Comment:
Any specific reason why we are doing this?

Reply:
CLI tool expects a comma-separated list of actions.
Users can give any text blob, so this sanitizes it.

Reply:
yeah, but would that lead to different results when they use the API directly? (if we are not sanitizing them in the API, I wonder if we should do it here - don't want to effect the response given from the API v/s using the toolkit

Reply:
Do you mean to give the same input for the same text blob? The tool should extract the action elements from the text blob if possible. The API expects an array of 'access' elements, we are just making it easier for users to provide that list here.